FlowCal/excel_ui.py

```diff
@@ -1126,23 +1181,24 @@ def generate_histograms_table(samples_table, samples, max_bins=1024):
             if pd.notnull(samples_table[header][sample_id]):
                 # Get units in which bins are being reported
                 unit = samples_table[header][sample_id]
-                # Decide whether to produce histograms in linear or log scale
-                if (unit != 'Channel') and \
-                    bool(sample.amplification_type(channel)[0]):
+                # Decide which scale to use
+                # Channel units result in linear scale. Otherwise, scale is
+                # decided based on amplification type.
+                if unit == 'Channel':
+                    scale = 'linear'
+                elif sample.amplification_type(channel)[0]:
                     scale = 'log'
                 else:
-                    scale = 'linear'
+                    scale = 'logicle'
                 # Define number of bins
                 nbins = min(sample.resolution(channel), max_bins)
-                # Calculate bin edges
-                bin_edges = sample.hist_bins(channel, nbins, scale)
-                # Calculate bin centers
-                if scale == 'linear':
-                    bin_centers = (bin_edges[:-1] + bin_edges[1:])/2
-                elif scale == 'log':
-                    bin_centers = (np.log10(bin_edges[:-1]) + \
-                        np.log10(bin_edges[1:]))/2
-                    bin_centers = 10**bin_centers
+                # Calculate bin edges and centers
+                # We generate twice the necessary number of bins. We then take
+                # every other value as the proper bin edges, and the remaining
+                # values as the bin centers.
+                bins_extended = sample.hist_bins(channel, 2*nbins, scale)
+                bin_edges = bins_extended[::2]
+                bin_centers = bins_extended[1::2]
```

Comment:
sorry, I'm thinking from the perspective of a user of the Excel UI. Bin centers are only ever calculated to output into the Histogram tab of the Excel UI, right? I'm thinking from that perspective as someone reading "Bin Centers" in the Histogram tab (not from a coder reading this block of code); they may expect "Bin Centers" to indicate the Euclidean midpoint. still think this method is fine, just that you might want to make sure it's clearly described to the end user who's using "Bin Centers"